podaac/forge_py/forge.py

```diff
@@ -1,64 +1,160 @@
+        - If the input is a single Polygon, it is returned without modification.
+        - If the input is a MultiPolygon, a new MultiPolygon containing only polygons
+          meeting the area threshold is returned.
+        - If entire area too small then return original multipolygon
+    """
+
+    # Ensure the geometry is a MultiPolygon for uniform processing
+    if isinstance(geometry, Polygon):
+        return geometry
+
+    # Filter polygons based on minimum area
+    filtered_polygons = [polygon for polygon in geometry.geoms if polygon.area >= min_area]
+
+    # Create a new MultiPolygon from the filtered polygons
+    result = MultiPolygon(filtered_polygons) if filtered_polygons else geometry
+
+    # Return WKT format if it was originally a string, otherwise return the geometry object
+    return result
+
+
+def generate_footprint(lon, lat, strategy=None, is360=False, path=None, **kwargs):
+    """
+    Generates a geographic footprint using a specified strategy.
+
+    Parameters:
+    ----------
+    lon, lat : list or array-like
+        Latitude and longitude values.
+    strategy : str, optional
+        The footprint strategy to use ('open_cv' or 'alpha_shape').
+    is360 : bool, default=False
+        If True, adjusts longitude values from 0-360 to -180-180 range.
+    path : str, optional
+        File path for saving output if the strategy requires it.
+    **kwargs : dict, optional
+        Additional parameters to be passed to the chosen strategy function.
+
+    Returns:
+    -------
+    str
+        The footprint as a WKT (Well-Known Text) string.
+    """
+    # Adjust longitude for 0-360 to -180-180 range if needed
     if is360:
-        lon_array = ((lon + 180) % 360.0) - 180
-    thinning = {'method': 'standard', 'value': thinning_fac}
-    alpha_shape = fit_footprint(lon_array, lat, alpha=alpha, thinning=thinning, cutoff_lat=cutoff_lat, smooth_poles=smooth_poles, fill_value=fill_value)
-    alpha_shape = alpha_shape.simplify(simplify)
+        lon = ((lon + 180) % 360.0) - 180
+
+    # Dispatch to the correct footprint strategy based on `strategy`
+    if strategy == "open_cv":
+        footprint = open_cv_footprint.footprint_open_cv(lon, lat, path=path, **kwargs)
+    else:
+        footprint = fit_footprint(lon, lat, **kwargs)
+        if not footprint.is_valid:
+            footprint = footprint.buffer(0)
+
+    if 'simplify' in kwargs:
+        footprint = footprint.simplify(tolerance=kwargs['simplify'], preserve_topology=True)
 
-    # If the polygon is not valid, attempt to fix self-intersections
-    if not alpha_shape.is_valid:
-        alpha_shape = alpha_shape.buffer(0)
+    # Optionally filter small polygons
+    if "min_area" in kwargs:
+        footprint = remove_small_polygons(footprint, kwargs['min_area'])
 
-    wkt_alphashape = dumps(alpha_shape)
-    return wkt_alphashape
+    return dumps(footprint, trim=True)
 
 
 def fit_footprint(
```

Comment:
This is only for alpha_shape right?  Can you move all the alpha_shape strategy to a new file?  And keep the forge.py small and simple.  Maybe you can create a new subdir with **strategy** and then put the open_cv_footprint.py and the alpha_shape_footprint.py in there to keep it more organized.  Sound good?